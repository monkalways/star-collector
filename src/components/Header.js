import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Button, Icon, Menu } from 'semantic-ui-react';

import { startLogout } from '../actions/auth';
import { menuChange } from '../actions/site';

export class Header extends React.Component {
    state = { activeItem: 'dashboard' }

    handleItemClick = (e, { name }) => this.props.menuChange(name);

    handleLogout = (e) => {
        this.props.startLogout();
    };

    render() {
        const { activeMenu } = this.props;
        return (
            <Menu color="blue" inverted fixed="top" size="huge">
                <Menu.Item header key="dashboard" name="star collector" to="/dashboard" as={Link} active={activeMenu === 'dashboard'} onClick={this.handleItemClick} />
                
                <Menu.Item position="right">
                    <Button onClick={this.handleLogout}>Logout</Button>
                </Menu.Item>
            </Menu>
        );
    }
}

const mapStateToProps = (state) => ({
    activeMenu: state.site.activeMenu
});

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout()),
    menuChange: (activeMenu) => dispatch(menuChange(activeMenu))
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);