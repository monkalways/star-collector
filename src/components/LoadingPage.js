import React from 'react';
import { Loader } from 'semantic-ui-react';

const containerStyle = {
    alignItems: 'center',
    display: 'flex',
    height: '100vh',
    justifyContent: 'center',
    width: '100vw'
};

const LoadingPage = () => {
    return (
        <div style={containerStyle}>
            <Loader active inline='centered' size="large">
                Loading
            </Loader>
        </div> 
    );
}

export default LoadingPage;