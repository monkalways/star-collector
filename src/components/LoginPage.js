import React from 'react';
import { connect } from 'react-redux';
import { Button, Dropdown, Form, Header, Icon, Segment } from 'semantic-ui-react';

import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => {

    return (
        <div className="box-layout">
            <div className="box-layout__box">
                <Segment padded basic>
                    <Header as="h1" textAlign="center">
                        <Header.Content>
                            <Icon name="star" color="yellow" />Star Collector
                        </Header.Content>
                    </Header>
                    <p>It's time to collect your stars.</p>
                    <Form>
                        <Button type='button' primary onClick={startLogin}>Login with Google</Button>
                    </Form>
                </Segment>
            </div>
        </div>
        
    );
}

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);