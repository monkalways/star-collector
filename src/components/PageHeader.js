import React from 'react';
import { Segment } from 'semantic-ui-react';

const pageHeaderStyle = {
    background: '#f7f7f7'
};

const PageHeader = (props) => {
    return (
        <Segment padded style={pageHeaderStyle}>
            {props.children}
        </Segment>
    );
}

export default PageHeader;