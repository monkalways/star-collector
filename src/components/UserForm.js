import React from 'react';
import { Button, Checkbox, Form, Input, Message, TextArea } from 'semantic-ui-react';

export default class UserForm extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            id: props.user ? props.user.id : '',
            stars: props.user ? props.user.stars : 0,
            halfStar: props.user ? props.user.halfStar : false,
            error: undefined
        };
    }

    onStarsChange = (e, data) => {
        const stars = data.value;
        this.setState(() => ({ stars }));
    };
    onHalfStarChange = (e, data) => {
      const halfStar = data.checked;
      this.setState(() => ({ halfStar }));
    };
    onSubmit = (e) => {
        e.preventDefault();

        if(!this.state.stars) {
            this.setState(() => ({ error: 'Please provide stars' }));
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                id: this.state.id,
                stars: this.state.stars,
                halfStar: this.state.halfStar
            });
        }
    };
    render() {
        return (
            <Form onSubmit={this.onSubmit}>
              
              {this.state.error && (
                <Message negative>
                  <Message.Header>{this.state.error}</Message.Header>
                </Message>
              )}
              <Form.Field>
                  <label>Stars</label>
                  <Input 
                    type="number"
                    placeholder="Stars"
                    autoFocus
                    value={this.state.stars}
                    onChange={this.onStarsChange}
                  />
              </Form.Field>
              <Form.Field>
                  <Checkbox 
                    label='Half Star' 
                    checked={this.state.halfStar}
                    onChange={this.onHalfStarChange} />
              </Form.Field>
              <Button type="submit" primary>Save</Button>
            </Form>
        );
    }
}
