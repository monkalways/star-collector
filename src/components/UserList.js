import React from 'react';
import { Card } from 'semantic-ui-react';

import UserListItem from './UserListItem';

const UserList = ({ users }) => {
  return (
    <Card.Group itemsPerRow="2">
      {users.map((user, index) => (<UserListItem key={index} {...user} />))}
    </Card.Group>
  );
}

export default UserList;