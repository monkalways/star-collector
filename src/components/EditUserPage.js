import React from 'react';
import { connect } from 'react-redux';

import PageHeader from './PageHeader';
import { startEditUser } from '../actions/users';
import UserForm from './UserForm';

export const EditUserPage = ({ user, handleEditUser, history }) => {

  const onSubmit = (user) => {
      handleEditUser(user);
      history.push('/');
  };

  return (
      <div>
          <PageHeader>
              <h1>Edit User {user.name}</h1>
          </PageHeader>
          <UserForm
              user={user}
              onSubmit={onSubmit}
          />
      </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
      user: state.users.find(user => user.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch) => ({
  handleEditUser: (user) => dispatch(startEditUser(user.id, user))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditUserPage);