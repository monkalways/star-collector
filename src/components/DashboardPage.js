import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Button, Dropdown, Form, Header, Icon, Segment } from 'semantic-ui-react';
import { push } from 'react-router-redux';

import PageHeader from './PageHeader';
import UserList from './UserList';

const pageHeaderTitleStyle= {
    fontWeight: 300,
};

const highlightStyle = {
    fontWeight: 700
};

const marginButtonStyle = {
    marginBottom: '2rem'
};

const DashboardPage = ({users}) => {
    return (
        <div>
            <div style={marginButtonStyle}>
                <PageHeader >
                    <h1 style={pageHeaderTitleStyle}>Who has more stars?</h1>
                </PageHeader>
            </div>
            <UserList users={users} />
        </div>
    )
};

const mapStateToProps = (state) => ({
    users: state.users
});

export default connect(mapStateToProps)(DashboardPage);