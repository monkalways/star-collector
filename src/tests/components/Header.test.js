import React from 'react';
import { shallow } from 'enzyme';

import { Header } from '../../components/Header';

let wrapper, startLogout;

beforeEach(() => {
    startLogout = jest.fn();
    wrapper = shallow(<Header startLogout={startLogout} />);
});

test('should render Header', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogout on button click', () => {
    wrapper.find('Button').prop('onClick')();
    expect(startLogout).toHaveBeenCalled();
});