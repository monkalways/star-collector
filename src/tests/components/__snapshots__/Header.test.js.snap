// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header 1`] = `
<Menu
  color="blue"
  fixed="top"
  inverted={true}
  size="huge"
>
  <MenuItem
    active={false}
    as={[Function]}
    header={true}
    key="dashboard"
    name="boilerplate"
    onClick={[Function]}
    to="/dashboard"
  />
  <MenuItem
    position="right"
  >
    <Button
      as="button"
      onClick={[Function]}
    >
      Logout
    </Button>
  </MenuItem>
</Menu>
`;
